packages/auth/src/grant/routes.ts

```diff
@@ -253,3 +255,76 @@ async function continueGrant(
     }
   }
 }
+
+// eslint-disable @typescript-eslint/no-explicit-any
+function isAccessTokenUpdateParameter(
+  param: any
+): param is {
+  access: AccessRequest[]
+} {
+  return (
+    param &&
+    param.access &&
+    Array.isArray(param.access) &&
+    param.access.length > 0 &&
+    param.access.every((a: any) => isAccessRequest(a))
+  )
+}
+// eslint-enable
+
+async function updateGrant(
+  deps: ServiceDependencies,
+  ctx: AppContext
+): Promise<void> {
+  // TODO: httpsig validation
+  const { continueId } = ctx.params
+  const continueToken = (ctx.headers['authorization'] as string)?.split(
+    'GNAP '
+  )[1]
+  const {
+    interact_ref: interactRef,
+    access_token: accessToken
+  } = ctx.request.body
+
+  if (
+    !continueId ||
+    !continueToken ||
+    !interactRef ||
+    !isAccessTokenUpdateParameter(accessToken)
+  ) {
+    ctx.status = 401
+    ctx.body = {
+      error: 'invalid_request'
+    }
```

Comment:
Should any of this be specified in the [OpenAPI spec](https://github.com/interledger/open-payments/blob/main/auth-server-open-api-spec.yaml#L200) and so it can be validated using the OpenAPI validation middleware?